libs/books/feature/src/lib/book-search/book-grid/book-grid-item/book-grid-item.component.ts

```diff
@@ -12,14 +10,28 @@ import { Book } from '@tmo/shared/models';
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 
-export class BookGridItemComponent {
+export class BookGridItemComponent implements OnInit {
 
   @Input() book: ReadingListBook;
+  isFinished: boolean;
 
   constructor(
     private readonly store: Store
   ) {}
 
+  ngOnInit() {
+    this.setIsFinished(this.book);
```

Comment:
This is messy. You could set up a selector to get you a list of searched for books with the `finished` property already populated.

Reply:
the input `book` will now have the `finished` property after updating the `getAllBooks` selector to enrich books with the `finished` property.